MagicTrialGame/Services/DataLoading/RoomFactory.cs

```diff
@@ -0,0 +1,21 @@
+using MagicTrialGame.Models.Rooms;
+
+namespace MagicTrialGame.Models
+{
+    public class RoomFactory
+    {
+        public List<RoomData> CreateRooms(List<RiddleData> riddleDataList)
+        {
+            var rooms = new List<RoomData>();
+
+            foreach (var riddleData in riddleDataList)
+            {
+                RoomData room = new RoomData(riddleData);
+                rooms.Add(room);
+            }
```

Comment:
Muzeme zjednodusit pomoci `Select`: 

```suggestion
            var rooms = riddleDataList.Select(riddleData => new RoomData(riddleData)).ToList();
```